app/connectors/optout/ITSAStatusUpdateConnector.scala

```diff
@@ -57,16 +58,17 @@ class ITSAStatusUpdateConnector @Inject()(val http: HttpClient, val appConfig: F
         case Status.NO_CONTENT => OptOutUpdateResponseSuccess(correlationId)
         case Status.NOT_FOUND =>
           log.error(s"response status: ${response.status}")
-          OptOutUpdateResponseFailure.notFoundFailure(correlationId, buildRequestUrlWith(taxableEntityId))
+          OptOutUpdateResponseFailure.notFoundFailure(buildRequestUrlWith(taxableEntityId))
         case _ =>
           response.json.validate[OptOutUpdateResponseFailure].fold(
             invalid => {
-              log.error(s"Json validation error parsing itsa-status update response, error $invalid")
-              OptOutUpdateResponseFailure.defaultFailure(correlationId)
+              val invalidJsonString = Json.prettyPrint(JsError.toJson(invalid))
```

Comment:
Hmm... if the json fails to validate, could it also fail to parse as Json and so not be possible to pretty-print it? Would it be safer here to just log the whole response body?

Reply:
done

Reply:
Cool. :)

Reply:
Looks like this is done. :)